api/src/modules/impact/impact.service.ts

```diff
@@ -142,7 +146,7 @@ export class ImpactService {
           fetchSpecification,
         );
         impactTableDto.supplierIds = this.getIdsFromTree(
-          entitiesWithPagination.entities,
+          entitiesWithPagination.entities as ImpactTableEntityType[],
```

Comment:
Which is the error that you get to have to cast this, when you have made the entities prop be optionally one type or another?